submissions/usides/friends-app/js/main.js

```diff
@@ -0,0 +1,159 @@
+    filterHumans(currentState.filter);
+    sortHumans(currentState.sort);
+    updateCards(currentState.humans);
+    timer = null;
+  }, 500);
+};
+
+const handleOptionClick = function (e) {
+  const button = e.target.closest('button');
+  if (!button) return;
+  const { dataset } = button;
+  if (dataset.sort) {
+    const prevActiveSortBtn = document.querySelector(
+      'button[data-sort].active',
+    );
+    prevActiveSortBtn.classList.remove('active');
+    button.classList.add('active');
+    sortHumans(dataset.sort);
+  }
+  if (dataset.filter) {
+    if (currentState.filter === dataset.filter) {
+      button.classList.toggle('active');
+      filterHumans('none');
+    } else {
+      const prevActiveFilterBtn = document.querySelector(
+        'button[data-filter].active',
+      );
+      if (prevActiveFilterBtn) prevActiveFilterBtn.classList.remove('active');
+      button.classList.add('active');
+      filterHumans(dataset.filter);
+    }
+    sortHumans(currentState.sort);
+  }
+  updateCards(currentState.humans);
+};
+
+const filterHumans = function (filterOption) {
+  const humansFilteredByName = humansOriginState.filter((elem) =>
+    elem.name.first.toLowerCase().startsWith(currentState.searchValue),
+  );
+  currentState.humans =
+    filterOption === 'none'
+      ? humansFilteredByName
+      : humansFilteredByName.filter(({ gender }) => gender === filterOption);
+  currentState.filter = filterOption;
+};
+
+const sortByAge = (a, b) => (a.dob.age <= b.dob.age ? 1 : -1);
+const sortByName = (a, b) =>
+  a.name.first.toLowerCase() >= b.name.first.toLowerCase() ? 1 : -1;
+
+const sortHumans = function (sortOption) {
+  switch (sortOption) {
+    case 'a-z':
+      currentState.humans.sort((a, b) => sortByName(a, b));
+      break;
+
+    case 'z-a':
+      currentState.humans.sort((a, b) => sortByName(b, a));
+      break;
+
+    case 'young':
+      currentState.humans.sort((a, b) => sortByAge(b, a));
+      break;
+
+    case 'old':
+      currentState.humans.sort((a, b) => sortByAge(a, b));
+      break;
+  }
+
+  currentState.sort = sortOption;
+};
+
+const updateCards = function (arr) {
+  cardsHolder.innerHTML = '';
+  const fragment = document.createDocumentFragment();
+  arr.forEach((elem) => {
+    const { name, picture, dob, gender } = elem;
+    const newElem = document.createElement('div');
+    newElem.classList.add('card');
+    const template = `
+    <p class="u-name">${name.first} ${name.last}</p>
+    <div class="splitter">
+      <img src="${picture.large}" alt="User Photo">
+      <div class="data-group">
+        <p class="label">Age</p>
+        <p class="data">${dob.age}</p>
+        <p class="label">Gender</p>
+        <p class="data">${gender}</p>
+      </div>
+    </div>`;
+    newElem.innerHTML = template;
+    fragment.appendChild(newElem);
+  });
+  cardsHolder.appendChild(fragment);
+};
+
+const getHumans = function () {
+  return fetch(HUMANS_LINK)
+    .then((res) => {
```

Comment:
```suggestion
    .then(({status}) => {
```

Reply:
It's not a good idea to destruct **response** to **response status** here, because I need **response** object to run `json()` method later in this code.

Reply:
Agree!